src/trunk.jl

```diff
@@ -27,6 +28,20 @@ The keyword arguments may include
 - `verbose::Int = 0`: if > 0, display iteration information every `verbose` iteration.
 - `verbose_subsolver::Int = 0`: if > 0, display iteration information every `verbose_subsolver` iteration of the subsolver.
 
+## Callback
+
+The callback is called after each iteration.
+The expected signature of callback is `(nlp, solver, wks)`, and the output of the callback is ignored.
+In addition to all information in `nlp` and `solver`, you can also use the following information stored in `wks`:
+
+- `wks[:iter]`: current iteration.
+- `wks[:f]`: current objective function value.
+- `wks[:optimal]`: true if the algorithm has found an optimal solution.
+- `wks[:elapsed_time]`: elapsed time in seconds.
+- `wks[:tired]`: true if the algorithm has exhausted all iterations or maximum time.
+- `wks[:stalled]`: true if the algorithm has stagnated.
+- `wks[:user_stop]`: set to true to stop the algorithm.
+
```

Comment:
Do we also need to add a warning saying that modifying `wks` may alter the algorithm?

Reply:
I added something.